src/zftools/metagraph.pyx

```diff
@@ -31,14 +31,22 @@ from cpython.mem cimport (
 from zftools.fastqueue cimport FastQueueForBFS
 
 
+def verify_graph_is_compatible(sage_graph):
+    if sage_graph.is_directed():
+        raise ValueError("This is a directed graph.  Currently, directed graph zero forcing is not implemented.")
+    elif sage_graph.has_loops():
```

Comment:
[nitpick] We can change this `elif` to `if` since we raise under the above condition

Reply:
That's true.  I kind of dislike `elif` for some reason.  Maybe because it's only one character away from being "elf" or something.  I'll change it.